src/python/pants/engine/rules_test.py

```diff
@@ -517,7 +517,7 @@ def a_from_b(b: B) -> A:
                 Rules with errors: 1
 
                   {fmt_rule(a_from_b)}:
-                    No rule was available to compute B with parameter type SubA
+                    No rule was able to compute B. No installed rules return the type B: Is the rule that you're expecting to run registered? If that type should be provided from outside the rule graph, consider declaring RootRule(B).
```

Comment:
The first two clauses are redundant. How about simplifying to only "No installed rules return the type B. Is the rule that..."

Reply:
Also, this line makes the file really long. I know it must fit on one line, so consider extracting it out into a string variable and then using an f-string.

Reply:
The first two clauses are redundant in this case, but in some cases this will be:
```
No rule was able to compute Get(B, A). No installed rules return the type B: ...
```

I'll fix the line length in the test.